Microsoft.Toolkit.Win32.UI.XamlHost/DesktopWindowXamlSourceExtensions.cs

```diff
@@ -4,29 +4,31 @@
 
 using System;
 using System.Runtime.InteropServices;
-using windows = Windows;
+using WUX = Windows.UI.Xaml;
 
 namespace Microsoft.Toolkit.Win32.UI.XamlHost
 {
     /// <summary>
-    /// COM wrapper required to access native-only methods on <see cref="windows.UI.Xaml.Hosting.DesktopWindowXamlSource" />
+    /// COM wrapper required to access native-only methods on <see cref="WUX.Hosting.DesktopWindowXamlSource" />
     /// </summary>
-    public static class DesktopWindowXamlSourceExtensions
+    static partial class DesktopWindowXamlSourceExtensions
     {
         /// <summary>
-        /// Gets the <see cref="IDesktopWindowXamlSourceNative" /> interface from a <see cref="windows.UI.Xaml.Hosting.DesktopWindowXamlSource" /> instance.
+        /// Gets the <see cref="IDesktopWindowXamlSourceNative" /> interface from a <see cref="WUX.Hosting.DesktopWindowXamlSource" /> instance.
         /// </summary>
+        /// <typeparam name="TInterface">The interface to cast to</typeparam>
         /// <param name="desktopWindowXamlSource">The DesktopWindowXamlSource instance to get the interface from</param>
         /// <returns><see cref="IDesktopWindowXamlSourceNative" /> interface pointer</returns>
         /// <remarks>
         /// This interface is the only way to set DesktopWindowXamlSource's target window for rendering.
         /// </remarks>
-        public static IDesktopWindowXamlSourceNative GetInterop(this windows.UI.Xaml.Hosting.DesktopWindowXamlSource desktopWindowXamlSource)
+        public static TInterface GetInterop<TInterface>(this WUX.Hosting.DesktopWindowXamlSource desktopWindowXamlSource)
+            where TInterface : class
         {
             var win32XamlSourceIntPtr = Marshal.GetIUnknownForObject(desktopWindowXamlSource);
             try
             {
-                var win32XamlSource = Marshal.GetTypedObjectForIUnknown(win32XamlSourceIntPtr, typeof(IDesktopWindowXamlSourceNative)) as IDesktopWindowXamlSourceNative;
+                var win32XamlSource = Marshal.GetTypedObjectForIUnknown(win32XamlSourceIntPtr, typeof(TInterface)) as TInterface;
```

Comment:
Is there any reason to not always return the most recent version of the interface here?

Reply:
Yes, to support RS4 clients.

---
In reply to: [261451885](https://github.com/windows-toolkit/Microsoft.Toolkit.Win32/pull/85#discussion_r261451885) [](ancestors = 261451885)